pkg/rbac/userData.go

```diff
@@ -192,7 +194,29 @@ func (user *userData) getNamespacedResources(cache *Cache, ctx context.Context,
 		}
 	}
 
+	//take intersection of authorizaed user namespaces and managed clusters:
+	managedClusters := cache.shared.managedClusters
+	namespaceNames := make([]string, len(nsResourcesAuth))
+	for name := range nsResourcesAuth {
+		namespaceNames = append(namespaceNames, name)
```

Comment:
It should be namespaceNames[i] = name. Since you already have assigned length, you should not append.

Reply:
Done.